language_service/src/lib.rs

```diff
@@ -50,9 +51,15 @@ impl<'a> LanguageService<'a> {
     /// This should be called before any language service requests have been made
     /// for the document, typically when the document is first opened in the editor.
     /// It should also be called whenever the source code is updated.
-    pub fn update_document(&mut self, uri: &str, version: u32, text: &str) {
+    pub fn update_document(
+        &mut self,
+        uri: &str,
+        version: u32,
+        text: &str,
+        package_type: PackageType,
```

Comment:
Seems a little odd that it's per-document, rather than a setting on the entire package/compilation, but I guess until we have multi-file projects this can be a workaround for now.

Reply:
Yeah, this is definitely a workarond until we have packages.